sources/salesforce/helpers/records.py

```diff
@@ -0,0 +1,121 @@
+"""Salesforce source helpers"""
+
+import pendulum
+
+from typing import Optional, Iterable, Dict, Set, Any, cast
+
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+from simple_salesforce import Salesforce
+from dlt.common.typing import TDataItem
+
+
+from ..settings import IS_PRODUCTION
+
+
+def _process_record(
+    record: Dict[str, Any], date_fields: Set[str], api_type: str
+) -> Dict[str, Any]:
+    """
+    Process a single Salesforce record by removing attributes and converting date fields.
+
+    Args:
+        record: The record to process
+        date_fields: Set of field names that contain date/datetime values
+        api_type: Whether this record came from Bulk API (timestamps), or standard API (ISO strings)
+
+    Returns:
+        The processed record
+    """
+    # Strip out the attributes field
+    record.pop("attributes", None)
+
+    for field in date_fields:
+        if record.get(field):
+            if api_type == "bulk":
+                # Bulk API returns timestamps, convert to ISO 8601
+                record[field] = pendulum.from_timestamp(
+                    record[field] / 1000,
+                ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+            else:
+                # Standard API return ISO strings, ensure consistent format
+                if record[field]:
+                    # pendulum.parse can also return Date, Time or Duration
+                    # Salesforce date/datetime fields are always DateTime
+                    dt = cast(pendulum.DateTime, pendulum.parse(record[field]))
+                    record[field] = dt.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
so Z means UTC timezone. so we are converting all things to utc timezone here. is that intended?
not sure what would be a good solution here